### ISSUE TITLE
Documentation about new "file " TCA type incomplete

### ISSUE THREAD

**Issue by hannesbochmann** [TYPO3-Documentation/TYPO3CMS-Reference-TCA#835]
The new "file" TCA type has the property "foreign_field" which is not documented here -> https://docs.typo3.org/m/typo3/reference-tca/main/en-us/ColumnsConfig/Type/File/Index.html

The property is especially useful for image fields inside a flexform. So it would be good if not everybody needs to dig through the code to find that possibility or even get the impression changing the referenced field name is not possible at all. Maybe there are even more undocumented options.

### TYPO3 versions

12 and newer

**Comment by brotkrueml** [comment]
Thanks for opening an issue. However, the property "foreign_field" is not mentioned in the according changelog: https://docs.typo3.org/c/typo3/cms-core/main/en-us/Changelog/12.0/Feature-98479-NewTCATypeFile.html. In the example in the changelog the foreign_field is also gone.

Maybe you can give an example in flexform where this is needed? And then also confirm that this works in "raw" TCA?

**Comment by hannesbochmann** [comment]
I haven't tested it in raw TCA because usually it's not needed there. In flexforms it's different. If you have a content element which has a flexform the value is saved in the field pi_flexform. If your flexform contains several fields for file references you need to configure a unique foreign_field to tell the fields apart. Without the foreign_field all fields would have the field pi_flexform as referenced. At least in former TYPO3 versions this was the case. Maybe this has changed and new references will be saved different without the need for foreign_field. But at least in older projects with existing references this configuration option is crucial and it's good that it's in place.

**Comment by hannesbochmann** [comment]
I'm a little bit confused right now. I'm pretty sure using foreign_field in flexforms worked at some point in TYPO3 12. But on the other side "foreign_field" wouldn't even be the correct configuration to change fields. This would be "foreign_match_fields". So maybe I messed something up. Anyways I took a look at the code and it's not possible to change the current behavior which is to use the flexform fieldname for the reference. So the problem I described in earlier TYPO3 versions exists no longer. Older projects should work fine unless the fieldname in the flexform and the former match fields do not match.
